src/borg/testsuite/helpers.py

```diff
@@ -234,6 +234,12 @@ def test_bad_syntax(self):
             # this is invalid due to the 2nd colon, correct: 'ssh://user@host/path'
             Location('ssh://user@host:/path')
 
+    def test_omit_archive(self):
+        loc = Location('ssh://user@host:1234/some/path::archive')
+        loc_without_archive = loc.omit_archive()
+        assert loc_without_archive.archive == "archive"
```

Comment:
did you see this ^^^?

Reply:
So I checked this. When '::<archive'-name> is present in the Location string, `Locatoin.archive` attribute will contain the string after '::".

With 'Location('ssh://user@host:1234/some/path')', i.e. without '::<archive-name', `Location.archive` attributed is `None.`

So I added tests for both test cases: When archive name is present in string and when it's not.

Since only `Location.orig` is being used to raise appropriate exceptions, I only modified that attribute. If needed I can change the value of `Location.archive` when returning from `omit_archive()` method, but I hope it does not break anything.

Reply:
OK, so please set `.archive` to None within `omit_archive`.

Reply:
Done